src/workers/atlas_worker.js

```diff
@@ -8,20 +8,38 @@ This Source Code Form is “Incompatible With Secondary Licenses”, as defined
 */
 
 import PeriodicWorker from './periodic_worker';
+import ChallengeResolutionStrategy from './atlas_strategies/challenge_resolution_strategy';
 
 export default class AtlasWorker extends PeriodicWorker {
-  constructor(web3, dataModelEngine, challengesRepository, logger) {
-    super(5000, logger);
+  constructor(web3, dataModelEngine, challengesRepository, strategy, logger) {
+    super(strategy.challengePullingInterval, logger);
     this.web3 = web3;
     this.dataModelEngine = dataModelEngine;
+    this.strategy = strategy;
     this.challengesRepository = challengesRepository;
+    if (!(this.strategy instanceof ChallengeResolutionStrategy)) {
+      throw new Error('A valid strategy must be provided');
+    }
+  }
+
+  async resolve(bundle, challengeId) {
+    if (await this.strategy.shouldResolveChallenge(bundle)) {
```

Comment:
I would change the way the code is organised here. We currently have a call chain where `periodicWork` is calling `tryResolve` which is calling `resolve`. In the two later cases the the names are extremely confusing. The first is actually consulting the strategy is to try to download the bundle, and then tries to download. The second is consulting the strategy is to resolve a downloaded bundle and then resolves. I would rather move all the strategy consulting into `periodicWork` and from there call `tryToDownload` and `tryToResolve`